File: FamilyVacationPlanner/src/FamilyVacationPlanner.Core/Model/BudgetAggregate/VacationBudget.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace FamilyVacationPlanner.Core;

public class VacationBudget
{
    public Guid VacationBudgetId { get; set; }
    public Guid TripId { get; set; }
    public Trip? Trip { get; set; }
    public string Category { get; set; } = string.Empty;
    public decimal AllocatedAmount { get; set; }
    public decimal? SpentAmount { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: FamilyVacationPlanner/src/FamilyVacationPlanner.WebApp/projects/family-vacation-planner-shared/src/lib/models/vacation-budget.ts
export interface VacationBudget {
  vacationBudgetId: string;
  tripId: string;
  category: string;
  allocatedAmount: number;
  spentAmount?: number;
  createdAt: string;
}

export interface CreateVacationBudgetCommand {
  tripId: string;
  category: string;
  allocatedAmount: number;
  spentAmount?: number;
}

export interface UpdateVacationBudgetCommand {
  vacationBudgetId: string;
  tripId: string;
  category: string;
  allocatedAmount: number;
  spentAmount?: number;
}
